### PR TITLE
[Core][Geometries]Making PointLocalCoordinates a constant method

### LINKED ISSUE #2011: [Core][Geometries]PointLocalCoordinates is not a constant method
https://github.com/KratosMultiphysics/Kratos/blob/de9420b4f0b391e93cdae22d07f7df438336a9e2/kratos/geometries/geometry.h#L885

Related with #1890

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixes #2011

Additionally removing unused virtual methods (overrided)

### CODE REVIEW

File: kratos/geometries/hexahedra_interface_3d_8.h [hunk 1]
```diff
@@ -700,15 +700,15 @@ template<class TPointType> class HexahedraInterface3D8 : public Geometry<TPointT
     }
 
     /**
-     * Returns the local coordinates of a given arbitrary point
```
Comment (loumalouomega) [inline kratos/geometries/hexahedra_interface_3d_8.h:703]: The correct name of this commit is Adding const to interface geometries, but I got confused :P

File: kratos/geometries/quadrilateral_3d_4.h [hunk 2]
```diff
@@ -1539,8 +1537,8 @@ template<class TPointType> class Quadrilateral3D4
 
         // Starting with xi = 0
         rResult = ZeroVector( 3 );
-        array_1d<double, 2> DeltaXi( 2, 0.0 );
-	const array_1d<double, 3> zero_array(3, 0.0);
+        array_1d<double, 2> DeltaXi = ZeroVector( 2 );
```
Comment (philbucher) [inline kratos/geometries/quadrilateral_3d_4.h:1540]: thats for AMatrix right?

**Review by philbucher — APPROVED** [review]
approving since @KratosMultiphysics/technical-committee requested this
Please still wait a few days with merging until others had the chance to take a look

File: kratos/geometries/quadrilateral_3d_4.h (hunk 2 above)
Comment (loumalouomega) [reply kratos/geometries/quadrilateral_3d_4.h:1540]: Yep

**Pull request merged**